File: Tarea7 (Ordenamiento por Selección)/Tarea7.cpp
#include <iostream>
using namespace std;

void selection(int a[], int n) {
    for (int i = 0; i < n; i++) {
        int small = i;
        for (int j = i + 1; j < n; j++) {
            if (a[small] > a[j]) {
                small = j;
            }
        }
        swap(a[i], a[small]);
    }
}

void printArr(int a[], int n) {
    for (int i = 0; i < n; i++) {
        cout << a[i] << " ";
    }
    cout << endl;
}

int main() {
    int a[] = {65, 26, 13, 23, 12};
    int n = sizeof(a) / sizeof(a[0]);

    cout << "Arreglo antes de ser ordenado:" << endl;
    printArr(a, n);

    selection(a, n);

    cout << "Arreglo despues de ser ordenado:" << endl;
    printArr(a, n);

    return 0;
}

File: Tarea7 (Ordenamiento por Selección)/Tarea7.js
function selection(a) {
    for (let i = 0; i < a.length; i++) {
        let small = i;
        for (let j = i + 1; j < a.length; j++) {
            if (a[small] > a[j]) {
                small = j;
            }
        }
        [a[i], a[small]] = [a[small], a[i]]; // swap
    }
}

function printArr(a) {
    console.log(a.join(" "));
}

let a = [65, 26, 13, 23, 12];

console.log("Arreglo antes de ser ordenado:");
printArr(a);

selection(a);

console.log("Arreglo despues de ser ordenado:");
printArr(a);
